rust/src/lib.rs

```diff
@@ -38,26 +39,24 @@ impl InstrumentationContext {
         )
     }
 
-    pub fn enter(&mut self, fi: &FrameInfo) -> Result<()> {
+    pub fn enter(&mut self, func_index : u32, func_name : &str) -> Result<()> {
         let mut fc = FunctionCall {
-            index: fi.func_index(),
+            index: func_index,
             start: SystemTime::now(),
             end: SystemTime::now(),
             within: Vec::new(),
             raw_name: None,
             name: None,
         };
-        if let Some(name) = fi.func_name() {
-            fc.name = Some(demangle_function_name(name.to_string()));
-            fc.raw_name = Some(name.to_string());
-        };
+        fc.name = Some(demangle_function_name(String::from(func_name)));
+        fc.raw_name = Some(String::from(func_name));
```

Comment:
If the name and demangled name are always available when `FunctionCall` is created then `fc.name` and `fc.raw_name` probably don't need to be `Options` anymore.

Reply:
That's a great idea of pass `None` through instead of `"placeholder"`, I didn't realize we had the same situation as before. Fixed.